test/sql/types/uuid/test_uuid.test

```diff
@@ -0,0 +1,80 @@
+# name: test/sql/types/uuid/test_uuid.test
+# description: Test UUID type
+# group: [uuid]
+
+statement ok
+PRAGMA enable_verification
+
+statement ok
+CREATE TABLE IF NOT EXISTS uuid (u uuid);
+
+statement ok
+INSERT INTO uuid VALUES ('A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11'), (NULL), ('47183823-2574-4bfd-b411-99ed177d3e43'), ('{10203040506070800102030405060708}');
+
+# output is always in the standard form
+query I
+SELECT uuid 'A0EEBC99-9C0B-4EF8-BB6D-6BB9BD380A11';
+----
+a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11
+
+# there can be multiple -
+query I
+SELECT uuid 'a0ee-bc99-9c0b-4ef8-bb6d-6bb9-bd38-0a11';
+----
+a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11
+
+# braces need to match
+statement error
+SELECT uuid '{a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11';
+
+# char length should be 32
+statement error
+SELECT uuid '47183823-2574-4bfd-b411-99ed177d3e4';
+
+query I
+SELECT u FROM uuid ORDER BY u;
+----
+NULL
+a0eebc99-9c0b-4ef8-bb6d-6bb9bd380a11
+10203040-5060-7080-0102-030405060708
+47183823-2574-4bfd-b411-99ed177d3e43
+
+# need to convert to varchar to sort by the string format
+query I
+SELECT u FROM uuid ORDER BY u::varchar;
```

Comment:
Not sure if it's a better option to make this the default behavior? But seems the comparison_operator are based on physical_type, don't know how to do it.

Reply:
If the UUID is encoded in big endian format, the string order and the hugeint order should be identical. According to wikipedia [this is also the most common encoding format](https://en.wikipedia.org/wiki/Universally_unique_identifier#Encoding). Perhaps swap around the bytes inside the hugeint to fix this?

Reply:
After that is fixed I would try to generate many UUIDs of different ranges and verify that the order is consistent between the string ordering and the physical type ordering, e.g. exhaustively generating (`000000-0000, 010000-000, etc`).

Reply:
Also would like to see the same large set of UUIDs used to test for round-tripping (e.g. so that `'00000-0000..'::UUID::VARCHAR == '00000-0000..'` for all UUIDs in that list).

Reply:
You mean we generate this UUID set and commit to the repo? Or generate on the fly?

Reply:
Should I randomly generate 100-200 uuid and use them for the testing?

Reply:
Yes, that is fine. Perhaps these could also be generated from within SQL, e.g.:

```sql
create function hex_char(i) as substr('0123456789abcdef', (i%16)::int+1, 1);
create function hex_char_repeat(i, j, rep) as repeat(hex_char(i)||hex_char(j), rep);
create function uuid_generate(i, j) as concat(hex_char_repeat(i, j, 4), '-', hex_char_repeat(i, j, 2), '-', hex_char_repeat(i, j, 2), '-', hex_char_repeat(i, j, 2), '-', hex_char_repeat(i, j, 6));

select uuid_generate(i, j) from range(0,16) tbl(i), range(0, 16) tbl2(j);
```